cmd/clusterctl/client/cluster/upgrader.go

```diff
@@ -390,9 +418,90 @@ func (u *providerUpgrader) doUpgrade(upgradePlan *UpgradePlan) error {
 	return nil
 }
 
-func newProviderUpgrader(configClient config.Client, repositoryClientFactory RepositoryClientFactory, providerInventory InventoryClient, providerComponents ComponentsClient) *providerUpgrader {
+func (u *providerUpgrader) scaleDownProvider(provider clusterctlv1.Provider) error {
+	log := logf.Log
+	log.Info("Scaling down", "Provider", provider.Name, "Version", provider.Version, "Namespace", provider.Namespace)
+
+	cs, err := u.proxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Fetch all Deployments belonging to a provider.
+	deploymentList := &appsv1.DeploymentList{}
+	if err := cs.List(ctx,
+		deploymentList,
+		client.InNamespace(provider.Namespace),
+		client.MatchingLabels{
+			clusterctlv1.ClusterctlLabelName: "",
+			clusterv1.ProviderLabelName:      provider.ManifestLabel(),
+		}); err != nil {
+		return errors.Wrapf(err, "failed to list Deployments for provider %s", provider.Name)
+	}
```

Comment:
What about deployments in capi-webhook (that exists only when starting point is v1alpha3)

Reply:
I’m not sure if we should also scale them down
As they are only passive webhooks they don’t really lead to any problems 

So as long as the CRDs are not yet patched they will ensure that conversions for the old types still work and after the CRDs are patched they are just unused